src/Tool/RequiredParameterTrait.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+namespace League\OAuth2\Client\Tool;
+
+use BadMethodCallException;
+
+trait RequiredParameterTrait
+{
+    /**
+     * Checks for a required parameter in a hash.
+     *
+     * @throws BadMethodCallException
+     * @param  string $name
+     * @param  array  $params
+     * @return void
+     */
+    private function checkRequiredParameter($name, array $params)
```

Comment:
This same trait is useful for `AccessToken` and possibly other supporting classes, such as implementations of `UserInterface`.